### PR TITLE
Add Race component

### PR DESCRIPTION
**Pull request by daelynj**
This is the first MVP of an image that 'moves' relative to your typing
progress. The padding of the image is hard coded in css, I'm hoping for
some direction / a topic to look into that would better achieve this.

### CODE REVIEW

File: src/component/race/Race.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    return "position" + Math.floor(props.currentWordIndex / wordsToAdvance);
+  }
+  return (
+    <div className={calculatePosition()}>
+      <img src={octane} alt="Octane" />
+    </div>
+  );
+}
+
+export default Race;
```
Comment (jgayfer) [inline src/component/race/Race.js]: ⚒ Would rather we add the new component in a separate commit. Makes it easier to review the parts that touch existing code.

File: src/component/race/Race.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import octane from "./images/octane.png";
+import "./css/Race.css";
+
+function Race(props) {
+  const wordsToAdvance = Math.round(props.wordsLength / 15);
```
Comment (jgayfer) [inline src/component/race/Race.js]: What significance does 15 have?

File: src/component/Board.js [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
 function Board(props) {
+  function renderRace() {
+    return (
+      <Race
+        wordsLength={props.words.length}
+        currentWordIndex={props.currentWordIndex}
+      />
+    );
+  }
+
```
Comment (jgayfer) [inline src/component/Board.js]: ⚒ I'm not a super big fan of these functions that don't anything but render a component, but this is more-so my stylistic preference; I like to avoid extraneous functions unless they significantly improve readability, which I don't think is the case here.

This could be a separate PR if we choose to make the change, as we have a few other instances of this.

File: src/component/Game.js [hunk 4]
```diff
@@ -2,7 +2,11 @@ import React, { useState } from "react";
 import Board from "./Board";
 
 function Game() {
-  const [words] = useState("this is text".split(" "));
+  const [words] = useState(
+    "this is text that you are typing in a typing game really fast wow you are so fast at typing this is text that you are typing in a typing game really fast wow you are so fast at typing".split(
+      " "
+    )
+  );
```
Comment (jgayfer) [inline src/component/Game.js]: This should be in a separate commit as it doesn't directly relate to integrating the race component.

File: src/component/race/css/Race.css [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  padding-left: 26em;
+}
+
+.position14 {
+  padding-left: 28em;
+}
+
+.position15 {
+  padding-left: 30em;
+}
```
Comment (jgayfer) [inline src/component/race/css/Race.css]: As you pointed out, this isn't great.

Instead we could use in line styles to make things dynamic.

```
<div style={{ padding-left: `{<padding value>}` }} />
```

File: src/component/race/Race.js (hunk 2 above)
Comment (daelynj) [reply src/component/race/Race.js]: @jgayfer you can think of that number as 'positions' in the race. It could be saved into a variable; it's a limiter on how fast or slow you want the icon to move.

File: src/component/Board.js (hunk 3 above)
Comment (daelynj) [reply src/component/Board.js]: The change would be to do all of the rendering in the return statement of the `Board` component? I'm open to everything, as I don't know many React mannerisms, if it makes more sense not to have these extraneous functions then I'm happy to change it. We could even move everything into the `Game` component and delete `Board`.

File: src/component/Game.js (hunk 4 above)
Comment (daelynj) [reply src/component/Game.js]: @jgayfer this was accidentally added to the commit, it shouldn't have been included.

File: src/component/Board.js (hunk 3 above)
Comment (daelynj) [reply src/component/Board.js]: Will get to this in a different PR

File: src/component/race/Race.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import octane from "./images/octane.png";
+
+function Race(props) {
+  const racePositions = 15;
```
Comment (jgayfer) [inline src/component/race/Race.js]: I like this being in a constant, but doesn't seem right to hard code.

Comment (daelynj) [reply src/component/race/Race.js]: @jgayfer I think the reason for hard coding it right now is because there's no styled race design to have the image move inside of. Once there is something, the 'racePositions' so to speak will be set based on the size of the race box etc. Does that make sense?

Comment (jgayfer) [reply src/component/race/Race.js]: A more robust solution would be to use a percentage based style, such that we don't need to know the width of the container.
